sdks/python/apache_beam/transforms/core.py

```diff
@@ -1387,6 +1387,11 @@ def partition_for(self, element, num_partitions, *args, **kwargs):
     return self._fn(element, num_partitions, *args, **kwargs)
 
 
+def _check_fn_use_yield_and_return(fn):
+  source_code = inspect.getsource(fn)
+  return " yield " in source_code and " return " in source_code
```

Comment:
only check the function source code to check yield and return, which should cover most of cases.